core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -197,6 +167,48 @@ public void onError(Status error) {
     });
   }
 
+  @VisibleForTesting
+  static NameResolver getNameResolver(String target, NameResolver.Factory nameResolverFactory,
+      Attributes nameResolverParams) {
+    // Finding a NameResolver. Try using the target string as the URI. If that fails, try prepending
+    // "dns:///".
+    URI targetUri = null;
+    StringBuilder uriSyntaxErrors = new StringBuilder();
+    try {
+      targetUri = new URI(target);
+      // For "localhost:8080" this would likely return null, because "localhost" is parsed as the
+      // scheme. Will fall into the next branch and try "dns:///localhost:8080".
+    } catch (URISyntaxException e) {
+      // "foo.googleapis.com:8080" will trigger this exception, because "foo.googleapis.com" is an
```

Comment:
"foo.googleapis.com:8080" will not trigger the exception. Please change it to "127.0.0.1:8080".